unstructured/metrics/evaluate.py

```diff
@@ -187,3 +177,46 @@ def measure_element_type_accuracy(
     _write_to_file(export_dir, "all-docs-element-type-frequency.tsv", df)
     _write_to_file(export_dir, "aggregate-scores-element-type.tsv", agg_df)
     _display(agg_df)
+
+
+def group_text_extraction_accuracy(
+    grouping: str, data_input: Union[pd.DataFrame, str], export_dir: str
+) -> None:
+    """
+    Aggregates accuracy and missing metrics by 'doctype' or 'connector', exporting to TSV.
+
+    Args:
+        grouping (str): Grouping category ('doctype' or 'connector').
+        data_input (Union[pd.DataFrame, str]): DataFrame or path to a CSV/TSV file.
+        export_dir (str): Directory for the exported TSV file.
+    """
+    if grouping not in ("doctype", "connector"):
+        raise ValueError("Invalid grouping category. Returning a non-group evaluation.")
+    if isinstance(data_input, str):
+        if not os.path.exists(data_input):
+            raise FileNotFoundError(f"File {data_input} not found.")
+        if data_input.endswith(".csv"):
+            df = pd.read_csv(data_input)
+        elif data_input.endswith((".tsv", ".txt")):
+            df = pd.read_csv(data_input, sep="\t")
+        else:
+            raise ValueError("Please provide a .csv or .tsv file.")
+    else:
+        df = data_input
+    if df.empty or grouping not in df.columns or df[grouping].isnull().all():
+        raise SystemExit(
+            f"Data cannot be aggregated by `{grouping}`. \
+                         Check if it's empty or the column is missing/empty."
```

Comment:
In Python, two string literals that immediately follow each other are automatically concatenated. So this error message would be formatted:
```python
        raise SystemExit(
            f"Data cannot be aggregated by `{grouping}`."
            f" Check if it's empty or the column is missing/empty."
        )
```
Note that the blank space needed to separate the first fragment from the second is placed at the _beginning_ of the second (and any later) lines. This makes it easy to confirm by vertically scanning the aligned left of the block rather than the ragged right side. It's a common oversight to forget the separating space such that you get ".. grouping.Check if ..." in the output, so a convention that makes it easy to visually confirm the spaces is welcome :)

Btw, the output from the original version would look like this when it printed out:
"... by grouping.                                                  Check if it's empty ..."

This inclusion of leading space in continuation lines is avoided by using adjacent string concatenation.